h/services/group.py

```diff
@@ -102,6 +104,24 @@ def annotation_count(self, pubid):
             .filter_by(groupid=pubid, shared=True)
             .count())
 
+    def groupids_readable_by(self, user):
+        """
+        Return a list of pubids for which the user has read access.
+
+        If the passed-in user is ``None``, this returns the list of
+        world-readable groups.
+        """
+        readable = (Group.readable_by == ReadableBy.world)
+
+        if user is not None:
+            readable_member = sa.and_(Group.readable_by == ReadableBy.members, Group.members.any(User.id == user.id))
+            readable = sa.or_(readable, readable_member)
+
+        ids = [record[0] for record in self.session.query(Group.pubid).filter(readable)]
```

Comment:
Oh, I didn't know about this! I always thought it just returns a normal tuple.

Fixed.